Lib/openbakery/profiles/name.py

```diff
@@ -419,48 +419,55 @@ def com_google_fonts_check_name_match_familyname_fullfont(ttFont):
 
 
 @check(
-    id="com.adobe.fonts/check/postscript_name",
+    id="com.adobe.fonts/check/postscript_name_characters",
     proposal="https://github.com/miguelsousa/openbakery/issues/62",
 )
-def com_adobe_fonts_check_postscript_name(ttFont):
-    """PostScript name follows OpenType specification requirements?"""
+def com_adobe_fonts_check_postscript_name_characters(ttFont):
+    """PostScript name contains only allowed characters a-ZA-Z and hyphen?"""
     import re
     from openbakery.utils import get_name_entry_strings
 
-    bad_entries = []
+    bad_entry_count = 0
 
     # <Postscript name> may contain only a-zA-Z0-9
-    # and one hyphen
     bad_psname = re.compile("[^A-Za-z0-9-]")
     for string in get_name_entry_strings(ttFont, NameID.POSTSCRIPT_NAME):
         if bad_psname.search(string):
-            bad_entries.append(
-                {
-                    "field": "PostScript Name",
-                    "value": string,
-                    "rec": ("May contain only a-zA-Z0-9 characters and a hyphen."),
-                }
+            bad_entry_count += 1
+            yield FAIL, Message(
+                "bad-psname-characters",
+                f"PostScript name {string} contains disallowed characters.",
             )
+
+    if bad_entry_count == 0:
+        yield PASS, Message(
+            "psname-chracters-ok", "PostScript name contains only allowed characters."
```

Comment:
Typo in `chracters`.

I think that the tests didn't fail because `assert_PASS()` doesn't validate the error's string ID, whereas `assert_results_contain()` does.

Try changing the `assert_PASS()`es to `assert_results_contain()` to confirm this hypothesis. The former are also using `"psname-ok"` which is not an error string ID yielded by any of the checks now.

Reply:
Hypothesis confirmed and code updated. Thanks for catching this!

Reply:
Message codes are only meant to distinguish between different ways a check fails to pass.
I don't see the point in giving a message code to a PASS. I don't think there's any check that can PASS in different manners (or that there would be any benefit in having such a thing).

Reply:
> I don't see the point in giving a message code to a PASS

The purpose is that one gets the assurance that the check is yielding the expected result, no matter if it's PASS or otherwise.

Reply:
to give names to PASS results sounds like a distraction

Reply:
Thanks for sharing your opinion. Browsing the commit history I get the sense that Josh and I are not the only contributors that find value in assigning a message code to a PASS result.